src/index.js

```diff
@@ -1,390 +1,586 @@
+              childNode,
+              nextChildren[nextChStart],
+              lifecycle,
+              eventProxy,
+              isSvg
+            )
+            nextKeyed[nextKey] = true
+            lastChStart++
+          } else {
+            if ((savedNode = lastKeyed[nextKey]) != null) {
+              patchElement(
+                element,
+                element.insertBefore(
+                  savedNode.element,
+                  childNode && childNode.element
+                ),
+                savedNode,
+                nextChildren[nextChStart],
+                lifecycle,
+                eventProxy,
+                isSvg
+              )
+              nextKeyed[nextKey] = true
+            } else {
+              patchElement(
+                element,
+                childNode && childNode.element,
+                null,
+                nextChildren[nextChStart],
+                lifecycle,
+                eventProxy,
+                isSvg
+              )
+            }
+          }
+          nextChStart++
         }
-      } else if (name in element && name !== "list" && !isSvg) {
-        element[name] = value == null ? "" : value
-      } else if (value != null && value !== false) {
-        element.setAttribute(name, value)
       }
 
-      if (value == null || value === false) {
-        element.removeAttribute(name)
+      while (lastChStart <= lastChEnd) {
+        if (getKey((childNode = lastChildren[lastChStart++])) == null) {
+          removeElement(element, childNode)
+        }
+      }
+
+      for (var key in lastKeyed) {
+        if (nextKeyed[key] == null) {
+          removeElement(element, lastKeyed[key])
+        }
       }
     }
   }
 
-  function createElement(node, isSvg) {
-    var element =
-      typeof node === "string" || typeof node === "number"
-        ? document.createTextNode(node)
-        : (isSvg = isSvg || node.nodeName === "svg")
-          ? document.createElementNS(
-              "http://www.w3.org/2000/svg",
-              node.nodeName
-            )
-          : document.createElement(node.nodeName)
-
-    var attributes = node.attributes
-    if (attributes) {
-      if (attributes.oncreate) {
-        lifecycle.push(function() {
-          attributes.oncreate(element)
-        })
-      }
+  return (nextNode.element = element)
+}
 
-      for (var i = 0; i < node.children.length; i++) {
-        element.appendChild(
-          createElement(
-            (node.children[i] = resolveNode(node.children[i])),
-            isSvg
-          )
-        )
-      }
+var createVNode = function(name, props, children, element, key, type) {
+  return {
+    name: name,
+    props: props,
+    children: children,
+    element: element,
+    key: key,
+    type: type
+  }
+}
 
-      for (var name in attributes) {
-        updateAttribute(element, name, attributes[name], null, isSvg)
-      }
-    }
+var createTextVNode = function(text, element) {
```

Comment:
If you use ES6 default parameter and organise params like below then createTextVNode is not needed at all:
```js
var createVNode = function(name, element, props={}, children=[],  key=null, type=TEXT_NODE) {
 ...
};
```

Reply:
Please read [this](https://github.com/hyperapp/hyperapp/blob/master/CONTRIBUTING.md#style) regarding which ES spec we follow.